src/obs-progress.cpp

```diff
@@ -125,12 +121,19 @@ void updateSceneInfo()
 {
 	try
 	{
-		obs_source_t* currentSceneSource = obs_frontend_get_current_scene(); // This is the only call that increments the count, so we have to release it
-		const char* name = obs_source_get_name(currentSceneSource);
+		// Clean up the previous source/scenes
+		_sources.clear();
+		obs_source_release(_currentSceneSource);
+		_progressDockWidget->clearProgressBars();
 
-		obs_scene_t* currentScene = obs_scene_from_source(currentSceneSource);
-		obs_source_release(currentSceneSource);
+		// Retrieve the current scene
+		_currentSceneSource = obs_frontend_get_current_scene(); // This is the only call that increments the count, so we have to release it
+		const char* name = obs_source_get_name(_currentSceneSource);
 
+		obs_scene_t* currentScene = obs_scene_from_source(_currentSceneSource);
+		obs_source_release(_currentSceneSource);
```

Comment:
You shouldn't need to do this here anymore, because we'll do it when we next update the scene info (or stop our timer).